SPDY/SPDYCacheStoragePolicy.m

```diff
@@ -13,6 +13,59 @@
 
 #import "SPDYCacheStoragePolicy.h"
 
+typedef struct _HTTPTimeFormatInfo {
+    const char *readFormat;
+    const char *writeFormat;
+    BOOL usesHasTimezoneInfo;
+} HTTPTimeFormatInfo;
+
+static HTTPTimeFormatInfo kTimeFormatInfos[] =
+{
+    { "%a, %d %b %Y %H:%M:%S %Z", "%a, %d %b %Y %H:%M:%S GMT", YES }, // Sun, 06 Nov 1994 08:49:37 GMT  ; RFC 822, updated by RFC 1123
+    { "%A, %d-%b-%y %H:%M:%S %Z", "%A, %d-%b-%y %H:%M:%S GMT", YES }, // Sunday, 06-Nov-94 08:49:37 GMT ; RFC 850, obsoleted by RFC 1036
+    { "%a %b %e %H:%M:%S %Y", "%a %b %e %H:%M:%S %Y", NO },           // Sun Nov  6 08:49:37 1994       ; ANSI C's asctime() format
+};
+
+
+static NSDate *HTTPDateFromString(NSString *string)
+{
+    NSDate *date = nil;
+    if (string) {
+        struct tm parsedTime;
+        const char *utf8String = [string UTF8String];
+
+        for (int format = 0; (size_t)format < (sizeof(kTimeFormatInfos) / sizeof(kTimeFormatInfos[0])); format++) {
+            HTTPTimeFormatInfo info = kTimeFormatInfos[format];
+            if (info.readFormat != NULL && strptime(utf8String, info.readFormat, &parsedTime)) {
+                NSTimeInterval ti = (info.usesHasTimezoneInfo ? mktime(&parsedTime) : timegm(&parsedTime));
+                date = [NSDate dateWithTimeIntervalSince1970:ti];
+                if (date) {
+                    break;
+                }
+            }
+        }
+    }
+
+    return date;
+}
+
+NSDictionary *HTTPCacheControlParameters(NSString *cacheControl)
+{
+    if (cacheControl.length == 0) {
+        return nil;
+    }
+
+    NSArray *components = [cacheControl componentsSeparatedByString:@","];
+    NSMutableDictionary *parameters = [NSMutableDictionary dictionaryWithCapacity:components.count];
+    for (NSString *component in components) {
+        NSArray *pair = [component componentsSeparatedByString:@"="];
+        NSString *key = [pair[0] stringByTrimmingCharactersInSet:[NSCharacterSet whitespaceAndNewlineCharacterSet]];
+        NSString *value = pair.count == 2 ? [pair[1] stringByTrimmingCharactersInSet:[NSCharacterSet whitespaceAndNewlineCharacterSet]] : @"";
```

Comment:
This is not right parsing. by RFC definition, `Cache-Control` is defined like

>    Cache-Control   = 1#cache-directive
>    cache-directive = token [ "=" ( token / quoted-string ) ]

and `quoted-string` can contains `=`. We must like concatenating `pair[1]` to last one with `=`.